components/Rating/rating.js

```diff
@@ -0,0 +1,33 @@
+const ratingStars = document.querySelectorAll(".rating input");
+const form = document.querySelector(".submit-review");
+const showReview = document.querySelector(".show-review");
+
+function checkStars(idx) {
+  if (ratingStars[idx].checked) {
+    for (let j = idx - 1; j >= 0; j--) {
+      ratingStars[j].checked = true;
+    }
+  } else {
+    for (let j = idx + 1; j < ratingStars.length; j++) {
+      ratingStars[j].checked = false;
+    }
+  }
+}
+
+export function rating() {
+  ratingStars.forEach((star, id) => {
+    star.addEventListener("change", () => checkStars(id));
+  });
+
+  form.addEventListener("submit", (e) => {
```

Comment:
The value of `form` will be `null` when the page is not loaded or the user is not on the same page, it might be a good idea to check for `null` before adding an `event listener`

Reply:
thanks, on it!